src/response/error.rs

```diff
@@ -4,108 +4,124 @@ use tonic::codegen::http;
 
 use crate::auth::AuthError;
 
+#[derive(Debug)]
+pub enum MomentoErrorCode {
+    /// Invalid argument passed to Momento client
+    InvalidArgumentError,
+    /// Service returned an unknown response
+    UnknownServiceError,
+    /// Resource with specified name already exists
+    AlreadyExistsError,
+    /// Cache with specified name doesn't exist
+    NotFoundError,
+    /// An unexpected error occurred while trying to fulfill the request
+    InternalServerError,
+    /// Insufficient permissions to perform operation
+    PermissionError,
+    /// Invalid authentication credentials to connect to service
+    AuthenticationError,
+    /// Request was cancelled by the server
+    CancelledError,
+    /// Request rate, bandwidth, or object size exceeded the limits for the account
+    LimitExceededError,
+    /// Request was invalid
+    BadRequestError,
+    /// Client's configured timeout was exceeded
+    TimeoutError,
+    /// Server was unable to handle the request
+    ServerUnavailable,
+    /// A client resource (most likely memory) was exhausted
+    ClientResourceExhausted,
+    /// System is not in a state required for the operation's execution
+    FailedPreconditionError,
+    /// Unknown error has occurred
+    UnknownError,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("{details}")]
+pub struct MomentoGrpcErrorDetails {
+    pub code: tonic::Code,
+    pub details: Cow<'static, str>,
+    pub metadata: Option<tonic::metadata::MetadataMap>,
```

Comment:
adding metadata produces errors like:

```
---- tests::ttl_validation stdout ----
thread 'tests::ttl_validation' panicked at tests/integration_test.rs:83:14:
failed to create cache: LimitExceeded(SdkError { message: "Request rate, bandwidth, or object size exceeded the limits for this account.  To resolve this error, reduce your usage as appropriate or contact us at support@momentohq.com to request a limit increase", error_code: LimitExceededError, inner_exception: Some(TonicStatus(Status { code: ResourceExhausted, message: "Your account has a limit of 10 caches. Please contact support if your use case necessitates more caches.", metadata: MetadataMap { headers: {"content-type": "application/grpc", "server": "awselb/2.0", "date": "Fri, 22 Mar 2024 17:04:33 GMT", "content-length": "0", "access-control-expose-headers": "grpc-status,grpc-message,grpc-encoding,grpc-accept-encoding", "vary": "origin", "vary": "access-control-request-method", "vary": "access-control-request-headers", "access-control-allow-credentials": "true"} }, source: None })), details: Some(MomentoGrpcErrorDetails { code: ResourceExhausted, details: "resource exhausted", metadata: Some(MetadataMap { headers: {"content-type": "application/grpc", "server": "awselb/2.0", "date": "Fri, 22 Mar 2024 17:04:33 GMT", "content-length": "0", "access-control-expose-headers": "grpc-status,grpc-message,grpc-encoding,grpc-accept-encoding", "vary": "origin", "vary": "access-control-request-method", "vary": "access-control-request-headers", "access-control-allow-credentials": "true"} }) }) })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Can remove the metadata if it seems like too much

Reply:
I think that is okay! We're giving people lots of ways to access more precise bits of info from the error if they want to.